src/layers/xr-layer/xr-layer-fragment.js

```diff
@@ -61,26 +61,24 @@ vec3 rgb2hsv(vec3 rgb) {
  }
 
 void main() {
-  float channel0Color = float(texture(channel0, vTexCoord).r) / sliderValues[0];
-  float channel1Color = float(texture(channel1, vTexCoord).r) / sliderValues[1];
-  float channel2Color = float(texture(channel2, vTexCoord).r) / sliderValues[2];
-  float channel3Color = float(texture(channel3, vTexCoord).r) / sliderValues[3];
-  float channel4Color = float(texture(channel4, vTexCoord).r) / sliderValues[4];
-  float channel5Color = float(texture(channel5, vTexCoord).r) / sliderValues[5];
+  float channel0Color = (float(texture(channel0, vTexCoord).r) - sliderValues[0][0]) / sliderValues[0][1];
+  float channel1Color = (float(texture(channel1, vTexCoord).r) - sliderValues[1][0]) / sliderValues[1][1];
+  float channel2Color = (float(texture(channel2, vTexCoord).r) - sliderValues[2][0]) / sliderValues[2][1];
+  float channel3Color = (float(texture(channel3, vTexCoord).r) - sliderValues[3][0]) / sliderValues[3][1];
+  float channel4Color = (float(texture(channel4, vTexCoord).r) - sliderValues[4][0]) / sliderValues[4][1];
+  float channel5Color = (float(texture(channel5, vTexCoord).r) - sliderValues[5][0]) / sliderValues[5][1];
```

Comment:
You want a ramp function here, right? Do we want to explicitly clip values that are too high or too low? That would be an easy thing to pull out as a util, with some unit tests.

Reply:
WebGL convention appears to be to just let it clip where it needs (the ramp function is the "default")